### ISSUE TITLE
[REVIEW]: liionpack: A Python package for simulating packs of batteries with PyBaMM

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#4051]
**Submitting author:** @TomTranter (<a href="http://orcid.org/0000-0003-4721-5941">Thomas Tranter</a>)
**Repository:** <a href="https://github.com/pybamm-team/liionpack" target ="_blank">https://github.com/pybamm-team/liionpack</a>
**Version:** v0.3
**Editor:** @timtroendle
**Reviewers:** @EricaEgg, @mefuller, @yangbai90
**Archive:** <a href="https://doi.org/10.5281/zenodo.6123712" target="_blank">10.5281/zenodo.6123712</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/1187d2a3788a6751cd4cce3dc016b7f3/status.svg)](https://joss.theoj.org/papers/1187d2a3788a6751cd4cce3dc016b7f3)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/1187d2a3788a6751cd4cce3dc016b7f3"><img src="https://joss.theoj.org/papers/1187d2a3788a6751cd4cce3dc016b7f3/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/1187d2a3788a6751cd4cce3dc016b7f3/status.svg)](https://joss.theoj.org/papers/1187d2a3788a6751cd4cce3dc016b7f3)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/pybamm-team/liionpack" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@EricaEgg & @mefuller, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @timtroendle know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @EricaEgg
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/pybamm-team/liionpack">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/pybamm-team/liionpack">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@TomTranter) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] #4141
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @mefuller
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/pybamm-team/liionpack">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/pybamm-team/liionpack">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@TomTranter) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @yangbai90
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/pybamm-team/liionpack">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/pybamm-team/liionpack">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@TomTranter) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @EricaEgg, @mefuller it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
Wordcount for `paper.md` is 1440

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.14 s (466.3 files/s, 64094.9 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                          36            562            838           2776
Markdown                         7            216              0            487
Jupyter Notebook                 9              0           2998            464
YAML                             9             41             25            354
JSON                             1             22              0            138
TeX                              1              9              0             89
JavaScript                       1              1              2             25
INI                              1              2              0             19
CSS                              1              0              0              4
-------------------------------------------------------------------------------
SUM:                            66            853           3863           4356
-------------------------------------------------------------------------------


Statistical information for the repository '3bb65ad5f6545c3f6a38a483' was
gathered on 2022/01/12.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Ferran Brosa Planell             5           135             92            0.69
Gavin Wiggins                   28           959            744            5.18
Priyanshu Agarwal                4            79             25            0.32
Robert Timms                     4           189            173            1.10
Saransh                         12           324            396            2.19
Tom Tranter                    200         16418          12703           88.50
TomTranter                       1            16              9            0.08
Valentin Sulzer                 11           315            159            1.44
ksnvikrant                       1            38              0            0.12
tomtranter                       1            93             38            0.40

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Ferran Brosa Planell         10            7.4          1.2               20.00
Gavin Wiggins               219           22.8          1.1               23.74
Priyanshu Agarwal            51           64.6          0.7               13.73
Robert Timms                 47           24.9          1.5               48.94
Saransh                     168           51.9          0.8               10.12
Tom Tranter                3487           21.2          0.7                7.92
Valentin Sulzer             222           70.5          2.0               28.83
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.04051/joss.04051/10.21105.joss.04051.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.04051/joss.04051/10.21105.joss.04051.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- None

MISSING DOIs

- 10.1007/s12532-018-0139-4 may be a valid DOI for title: CasADi: a software framework for nonlinear optimization and optimal control
- 10.1149/osf.io/67ckj may be a valid DOI for title: Python battery mathematical modelling (PyBaMM)
- 10.1016/0010-4485(76)90063-4 may be a valid DOI for title: The modified nodal approach to network analysis
- 10.1149/1945-7111/aba44b may be a valid DOI for title: Probing heterogeneity in li-ion batteries with coupled multiscale models of electrochemistry and thermal transport using tomographic domains
- 10.1149/1945-7111/ab9050 may be a valid DOI for title: Development of experimental techniques for parameterization of multi-scale lithium-ion battery models
- 10.1149/1945-7111/abbce4 may be a valid DOI for title: A suite of reduced-order models of a single-layer lithium-ion pouch cell

INVALID DOIs

- None
```

**Comment by timtroendle** [comment]
@whedon remind @EricaEgg in 3 weeks

**Comment by whedon** [comment]
Reminder set for @EricaEgg in 3 weeks

**Comment by timtroendle** [comment]
@whedon remind @mefuller in 3 weeks

**Comment by whedon** [comment]
Reminder set for @mefuller in 3 weeks

**Comment by TomTranter** [comment]
@timtroendle I fixed those DOIs in a review branch that I haven't merged yet. If there are any other necessary code changes I will also make them here https://github.com/pybamm-team/liionpack/tree/JOSS-review

**Comment by whedon** [comment]
OK, @yangbai90 is now a reviewer

**Comment by timtroendle** [comment]
@whedon remind @yangbai90 in 3 weeks

**Comment by whedon** [comment]
Reminder set for @yangbai90 in 3 weeks

**Comment by whedon** [comment]
:wave: @EricaEgg, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @mefuller, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @EricaEgg, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @mefuller, please update us on how your review is going (this is an automated reminder).

**Comment by mefuller** [comment]
Some additional text is needed in the paper and I have yet to personally run thorough tests on the capability.
Otherwise I am nearly done.

**Comment by whedon** [comment]
:wave: @yangbai90, please update us on how your review is going (this is an automated reminder).

**Comment by mefuller** [comment]
I have completed my review and recommend the paper for acceptance

**Comment by yangbai90** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

```

**Comment by yangbai90** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1007/s12532-018-0139-4 is OK
- 10.5334/jors.309 is OK
- 10.1109/TCS.1975.1084079 is OK
- 10.1149/1945-7111/aba44b is OK
- 10.1149/1945-7111/ab9050 is OK
- 10.1149/1945-7111/abbce4 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by yangbai90** [comment]
@whedon check repository

**Comment by whedon** [comment]
Wordcount for `paper.md` is 1556

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.10 s (680.8 files/s, 90630.3 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                          36            562            838           2776
Markdown                         7            223              0            494
Jupyter Notebook                 9              0           2998            464
YAML                            12             41             25            447
JSON                             1             22              0            138
TeX                              1              9              0             95
JavaScript                       1              1              2             25
INI                              1              2              0             19
CSS                              1              0              0              4
-------------------------------------------------------------------------------
SUM:                            69            860           3863           4462
-------------------------------------------------------------------------------


Statistical information for the repository 'c5dcfc2d5e4e827133d44b1f' was
gathered on 2022/02/08.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Ferran Brosa Planell             5           135             92            0.69
Gavin Wiggins                   28           959            744            5.18
Priyanshu Agarwal                4            79             25            0.32
Robert Timms                     4           189            173            1.10
Saransh                         12           324            396            2.19
Tom Tranter                    200         16418          12703           88.50
TomTranter                       1            16              9            0.08
Valentin Sulzer                 11           315            159            1.44
ksnvikrant                       1            38              0            0.12
tomtranter                       1            93             38            0.40

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Ferran Brosa Planell         10            7.4          1.2               20.00
Gavin Wiggins               219           22.8          1.1               23.74
Priyanshu Agarwal            51           64.6          0.7               13.73
Robert Timms                 47           24.9          1.5               48.94
Saransh                     168           51.9          0.8               10.12
Tom Tranter                3487           21.2          0.7                7.92
Valentin Sulzer             222           70.5          2.0               28.83
```

**Comment by yangbai90** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.04051/joss.04051/10.21105.joss.04051.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.04051/joss.04051/10.21105.joss.04051.pdf) :page_facing_up: :point_left:

**Comment by yangbai90** [comment]
I have finished my review part, now it's ready for the next step.

**Comment by EricaEgg** [comment]
Great work!
I have completed my review. My only recommendation is that it was not clear in the documentation that LaTeX is required. Please add this to installation instructions.

**Comment by TomTranter** [comment]
Thanks all, @EricaEgg I have now updated the documentation

**Comment by EricaEgg** [comment]
> Thanks all, @EricaEgg I have now updated the documentation

Perfect! Looks great. I recommend this paper for acceptance.

**Comment by TomTranter** [comment]
@timtroendle are we waiting on anything else?

**Comment by timtroendle** [comment]
@TomTranter, all looks good to me! There is however a lot of unused whitespace in the current paper version. May I suggest you pull Figure 1 above the "Algorithm" section heading? That will give you a much better text flow and it will also remove one (unused) page from the paper.

Apart from that, could you please:
- [ ] Make a tagged release of your software, and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository)
- [ ] Check the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it). You may also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here.

I can then move forward with accepting the submission.

**Comment by TomTranter** [comment]
> v0.3
> 10.5281/zenodo.6123712

@timtroendle

**Comment by timtroendle** [comment]
@whedon set v0.3 as version

**Comment by timtroendle** [comment]
@whedon set 10.5281/zenodo.6123712 as doi

**Comment by timtroendle** [comment]
@whedon set 10.5281/zenodo.6123712 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.6123712" target="_blank">10.5281/zenodo.6123712</a> is the archive.

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1007/s12532-018-0139-4 is OK
- 10.5334/jors.309 is OK
- 10.1109/TCS.1975.1084079 is OK
- 10.1149/1945-7111/aba44b is OK
- 10.1149/1945-7111/ab9050 is OK
- 10.1149/1945-7111/abbce4 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2976

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2976, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
👋 @TomTranter - I'm the JOSS AEiC this week, who handles the remaining processing of your paper.

I see that the zenodo deposit has fairly different metadata than the paper.  The paper has 11 authors, the archive has 9. The paper authors don't have first names, only first initials, while the archive authors mostly have two names, except one who has three and one who only has one.  The titles of the paper and the archive are also quite different. Normally, we suggest that these be consistent.  Can you either make them consistent, or explain your thoughts on why they shouldn't be?

Also, the JOSS style is to use full names of authors, usually a first and last name, sometimes with one or more initials as well (as in the the [example paper](https://joss.readthedocs.io/en/latest/submitting.html#example-paper-and-bibliography)).

I'm now going to proofread the paper and references, and may have further comments.

**Comment by danielskatz** [comment]
I have the following suggested change in the paper: https://github.com/pybamm-team/liionpack/pull/142

In addition, can you define DAE when you first use it?

**Comment by TomTranter** [comment]
@danielskatz the two authors who have not contributed code have contributed review of the paper and also supervision of the projects that the code was developed under. Is it ok to keep them on the paper? I will work to correct the other issues you raised. Apologies for the inconsistency it was my first time using zenodo

**Comment by danielskatz** [comment]
If the other two authors contributed to the software, I would add them to the zenodo - I think it's intended to recognize contributors to the project, not just those who made commits in the repo.  But this is your decision.

**Comment by danielskatz** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2978

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2978, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1495789697540542468) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2979
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.04051](https://doi.org/10.21105/joss.04051)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by danielskatz** [comment]
Congratulations to @TomTranter and co-authors!!

And thanks to @EricaEgg, @mefuller, and @yangbai90 for reviewing, and @timtroendle for editing!
We couldn't do this without you!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.04051/status.svg)](https://doi.org/10.21105/joss.04051)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.04051">
  <img src="https://joss.theoj.org/papers/10.21105/joss.04051/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.04051/status.svg
   :target: https://doi.org/10.21105/joss.04051
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.04051/status.svg)](https://doi.org/10.21105/joss.04051)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
